internal/explorer/server.go

```diff
@@ -328,31 +334,51 @@ func Setup(router *mux.Router, redisServer string, gitCommit string, database db
 func updateCacheRoutine(a App) {
 	// time to just pass tests before manipulating data
 	time.Sleep(time.Minute * 10)
-
+	batchSize := 100
+	currentBatch := 0
 	for {
 		nodes, _, err := a.db.GetNodes(types.NodeFilter{}, types.Limit{})
 		if err != nil {
 			continue
 		}
-		for _, node := range nodes {
-			const cmd = "zos.statistics.get"
-			var result struct {
-				Total gridtypes.Capacity `json:"total"`
-				Used  gridtypes.Capacity `json:"used"`
-			}
-			err = a.rmb.Call(context.Background(), uint32(node.TwinID), cmd, nil, &result)
-			if err != nil {
-				errSet := a.db.SetNodeStatusCache(uint32(node.NodeID), "up")
+		var wg sync.WaitGroup
+		beforeUpdate := time.Now()
+		for idx := range nodes {
+			wg.Add(1)
+			currentBatch++
+			go func(a App, node db.Node) {
+				defer wg.Done()
+				err := rmbCall(a, node)
+				if err != nil {
+					errSet := a.db.SetNodeStatusCache(uint32(node.NodeID), statusDown)
```

Comment:
setting node status is done twice and also the error handling with just the status change. maybe it would be better to use a status variable and change its value if there is an error and set the status once?